src/main/java/org/opensearch/sdk/ExtensionRestResponse.java

```diff
@@ -23,71 +24,76 @@ public class ExtensionRestResponse extends BytesRestResponse {
      * @param content  A plain text response string.
-     * @param consumedParams  Parameters consumed by the handler.
      */
-    public ExtensionRestResponse(RestStatus status, String content, List<String> consumedParams) {
+    public ExtensionRestResponse(ExtensionRestRequest request, RestStatus status, String content) {
         super(status, content);
-        addConsumedParamHeader(consumedParams);
+        addConsumedHeaders(request.consumedParams(), request.isContentConsumed());
     }
 
     /**
      * Creates a new plain text response.
      *
+     * @param request  the REST request being responded to.
      * @param status  The REST status.
      * @param contentType  The content type of the response string.
      * @param content  A response string.
-     * @param consumedParams  Parameters consumed by the handler.
      */
-    public ExtensionRestResponse(RestStatus status, String contentType, String content, List<String> consumedParams) {
+    public ExtensionRestResponse(ExtensionRestRequest request, RestStatus status, String contentType, String content) {
         super(status, contentType, content);
-        addConsumedParamHeader(consumedParams);
+        addConsumedHeaders(request.consumedParams(), request.isContentConsumed());
     }
 
     /**
      * Creates a binary response.
      *
+     * @param request  the REST request being responded to.
      * @param status  The REST status.
      * @param contentType  The content type of the response bytes.
      * @param content  Response bytes.
-     * @param consumedParams  Parameters consumed by the handler.
      */
-    public ExtensionRestResponse(RestStatus status, String contentType, byte[] content, List<String> consumedParams) {
+    public ExtensionRestResponse(ExtensionRestRequest request, RestStatus status, String contentType, byte[] content) {
         super(status, contentType, content);
-        addConsumedParamHeader(consumedParams);
+        addConsumedHeaders(request.consumedParams(), request.isContentConsumed());
     }
 
     /**
      * Creates a binary response.
      *
+     * @param request  the REST request being responded to.
      * @param status  The REST status.
      * @param contentType  The content type of the response bytes.
      * @param content  Response bytes.
-     * @param consumedParams  Parameters consumed by the handler.
      */
-    public ExtensionRestResponse(RestStatus status, String contentType, BytesReference content, List<String> consumedParams) {
+    public ExtensionRestResponse(ExtensionRestRequest request, RestStatus status, String contentType, BytesReference content) {
         super(status, contentType, content);
-        addConsumedParamHeader(consumedParams);
+        addConsumedHeaders(request.consumedParams(), request.isContentConsumed());
     }
 
-    private void addConsumedParamHeader(List<String> consumedParams) {
+    private void addConsumedHeaders(List<String> consumedParams, boolean contentConusmed) {
```

Comment:
This name is too literal and kind of tripping me up, what about breaking this functionality apart into something like `markParameter(s)Consumed(String([]) param)`, `markContentConsumed()`, and finally `record/reportConsumedItems()`

Reply:
I'm not sure "mark...Consumed..." is right, because that's the behavior already happening inside the RestRequest class.  The purpose here is to add header keys and values to pass these already-marked items with the Response to leverage the convenience of the superclass, rather than creating a whole new transport action for them.

I was already thinking of moving this class in #165.  Can we move the rename discussion there so as not to hold up this PR?